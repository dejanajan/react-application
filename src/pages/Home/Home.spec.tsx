import React from "react";
import HomePage, { filterList } from "./Home.page";
import { Source } from '../../models/repos';
import renderer from "react-test-renderer";

test("Render Home Page", () => {
  const component = renderer
  .create(
    <HomePage/>
  );

  const tree = component.toJSON();  
  expect(tree).toMatchSnapshot();
});



const reposList: Source[] | any[] = [{id: 255900773, name: "Test"}, 
                 {id: 255900756, name: "Angular"}, 
                 {id: 255900759, name: "React"}
                ];

const filteredRepos: Source[] | any[] = [{id: 255900773, name: "Test"}];


test("Testing FilteredReposList", () => {
  expect(filterList(reposList, "Test")).toEqual(filteredRepos);
});