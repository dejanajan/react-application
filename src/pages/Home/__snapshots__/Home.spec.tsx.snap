// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Home Page 1`] = `
<div
  className="sc-AxirZ hPKnIf"
>
  <p>
    Loading ...
  </p>
</div>
`;
